lhotse/audio.py

```diff
@@ -38,37 +38,53 @@ class AudioSource:
 
     def load_audio(
             self,
-            offset_seconds: float = 0.0,
-            duration_seconds: Optional[float] = None,
+            offset: Seconds = 0.0,
+            duration: Optional[Seconds] = None,
     ) -> np.ndarray:
         """
         Load the AudioSource (both files and commands) with librosa,
         accounting for many audio formats and multi-channel inputs.
         Returns numpy array with shapes: (n_samples) for single-channel,
         (n_channels, n_samples) for multi-channel.
         """
-        assert self.type in ('file', 'command')
+        assert self.type in ('file', 'command', 'url')
 
+        # TODO: refactor when another source type is added
         source = self.source
+
         if self.type == 'command':
-            if offset_seconds != 0.0 or duration_seconds is not None:
+            if offset != 0.0 or duration is not None:
                 # TODO(pzelasko): How should we support chunking for commands?
                 #                 We risk being very inefficient when reading many chunks from the same file
                 #                 without some caching scheme, because we'll be re-running commands.
                 raise ValueError("Reading audio chunks from command AudioSource type is currently not supported.")
             source = BytesIO(run(self.source, shell=True, stdout=PIPE).stdout)
+            samples, sampling_rate = read_audio(source)
 
-        with warnings.catch_warnings():
-            warnings.simplefilter('ignore')
-            samples, sampling_rate = read_audio(source, offset=offset_seconds, duration=duration_seconds)
-
-        # explicit sanity check for duration as librosa does not complain here
-        if duration_seconds is not None:
+        elif self.type == 'url':
```

Comment:
The comment at line 33 also needs to be updated.

Reply:
Thanks!